dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/SamplesToVariantSourceProcessor.java

```diff
@@ -0,0 +1,127 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.dbsnpimporter.jobs.steps.processors;
+
+import org.springframework.batch.item.ItemProcessor;
+
+import uk.ac.ebi.eva.commons.core.models.Aggregation;
+import uk.ac.ebi.eva.commons.core.models.IVariantSource;
+import uk.ac.ebi.eva.commons.core.models.StudyType;
+import uk.ac.ebi.eva.commons.core.models.VariantSource;
+import uk.ac.ebi.eva.dbsnpimporter.models.Sample;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Packs a set of {@link Sample} under a {@link uk.ac.ebi.eva.commons.core.models.IVariantSource} object, adding some
+ * additional batch information.
+ */
+public class SamplesToVariantSourceProcessor implements ItemProcessor<List<Sample>, IVariantSource> {
+
+    public static final String DBSNP_BUILD_KEY = "dbsnp-build";
+
+    public static final String DBSNP_BATCH_KEY = "dbsnp-batch-id";
+
+    private final String dbsnpBuild;
+
+    private final String batchId;
+
+    public SamplesToVariantSourceProcessor(int dbsnpBuild, int batchId) {
+        this.dbsnpBuild = String.valueOf(dbsnpBuild);
+        this.batchId = String.valueOf(batchId);
+    }
+
+    @Override
+    public IVariantSource process(List<Sample> samples) throws Exception {
+        if (areSamplesFromMultipleBatches(samples)) {
+            throw new IllegalArgumentException("Samples must belong to a single batch");
+        }
+
+        if (isSampleDuplicated(samples)) {
+            throw new IllegalArgumentException("The batch contains duplicate samples");
+        }
+
+        // Study ID, file ID, study name, file name
+        String batchName = samples.get(0).getBatch();
+        String studyId = batchName;
+        String studyName = batchName;
+        String fileId = batchName;
+        String fileName = batchName;
+
+        Map<String, Integer> samplesPosition = getSamplesPosition(samples);
+
+        Map<String, Object> metadata = getMetadata();
+
+        // Build any kind of IVariantSource
+        return new VariantSource(fileId, fileName, studyId, studyName, StudyType.COLLECTION, Aggregation.NONE,
+                                 new Date(), samplesPosition, metadata, null);
+    }
+
+    private boolean areSamplesFromMultipleBatches(List<Sample> samples) {
+        String batchName = null;
+
+        for (Sample s : samples) {
```

Comment:
A stream operation will be less efficient but maybe easier  to read:

long numberOfBatches =  samples.stream().map(s -> s.getBatch()).distinct().count();

Anyway, the method name makes clear what this method does

Reply:
This is only going to be executed once so if @jmmut agrees it is more readable I will change it. I don't have a strong opinion on the matter.

Reply:
I don't have a strong opinion. I prefer not to waste time changing already good enough code.